client/python/datajunction/client.py

```diff
@@ -69,12 +69,17 @@ def common_dimensions(
     def common_metrics(
         self,
         dimensions: List[str],
-    ):
+    ) -> List[str]:  # pragma: no cover # Tested in integration tests
         """
         Return common metrics for a set of dimensions.
         """
-        # TODO  # pylint: disable=W0511
-        # Add this to the server APIs first.
+        query_params = [("node_type", models.NodeType.METRIC.value)]
+        for dim in dimensions:
+            query_params.append(("dimension", dim))
+        response = self._session.get(
+            f"/dimensions/common/?{urlencode(query_params)}",
+        )
+        return [metric["name"] for metric in response.json()]
```

Comment:
Is this another case where we need to update the server response type to just be a list of strings?

Reply:
I was considering this, but then it'll blow up our calls to the backend from the UI when we call this endpoint to display linked nodes on a dimension. Right now we just do this single call, which will return all the info it needs (i.e., display name, node type etc), but otherwise we'd have to make one call per node that it gets back from this endpoint. Of course, if we lean into GraphQL this problem goes away...

Reply:
I see, makes sense. Maybe we can put the graphql stuff back but for a subset of the endpoints, for example only for the GET endpoints.

Reply:
Yes, I was having the same conversation in my head as you two just above. My opinion would be to:
- on all APIs that may return all metrics, limit the type to list of names
- use/add extra APIs to retrieve the details of a metric/node if the client needs it
- for GraphQL possible have a separate set of APIs, caveat: I am not an expert in these (graphql) standards.

Reply:
I like that pattern and agree we should keep it as consistent as we can.